File: 面试题 01.06. 字符串压缩/面试题 01.06. 字符串压缩.py
class Solution:
    def compressString(self, S: str) -> str:
        n = len(S)
        if n<=2:
            return S
        last = S[0]
        count = 1
        out = ''
        for c in S[1:]:
            if c==last:
                count+=1
            else:
                out += last+str(count)
                last = c
                count = 1
        out += c+str(count)
        return out if len(out)<n else S
            

File: 面试题 01.06. 字符串压缩/面试题 01.06. 字符串压缩.go
package main

import "fmt"

func main() {

}

func compressString(S string) string {
	if len(S) <= 2 {
		return S
	}
	last := byte(S[0])
	count := 1
	res := ""
	for _, cc := range S[1:] {
		c := byte(cc)
		if c == last {
			count++
		} else {
			res += string(last) + fmt.Sprint(count)
			last = c
			count = 1
		}
	}
	res += string(last) + fmt.Sprint(count)
	if len(res) >= len(S) {
		return S
	}
	return res
}

// 感受：类型转换，rune和byte
// int转string不可以用string(int)->返回\u0001，要用fmt.Sprint(int)
